sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -872,7 +872,8 @@ class Analyzer(
           if (missingExpr.nonEmpty) {
             extractedExprBuffer += ne
           }
-          ne.toAttribute
+          // alias will be cleaned in the rule CleanupAliases
+          if (ne.isInstanceOf[Alias]) ne else ne.toAttribute
```

Comment:
Well, wait a second.  I would like to fix this in branch-1.6 so a minimal fix would be preferred (and we can refactor master later if there is a good reason).